android/src/main/java/com/reactcommunity/rndatetimepicker/CustomTimePickerDialog.java

```diff
@@ -0,0 +1,220 @@
+    final View textInputPicker = this.findViewById(textInputPickerId);
+
+    return textInputPicker != null && textInputPicker.hasFocus();
+  }
+
+  /**
+   * Corrects minute values if they don't align with minuteInterval
+   *
+   * If the picker is in text input mode, correction will be postponed slightly to let the user
+   * finish the input
+   *
+   * If the picker is not in text input mode, correction will take place immidiately
+   *
+   * @param view the picker's view
+   * @param hourOfDay the picker's selected hours
+   * @param realMinutes 'real' minutes (0-59)
+   */
+  private void correctEnteredMinutes(final TimePicker view, final int hourOfDay, final int realMinutes) {
+    if (pickerIsInTextInputMode()) {
+      final EditText textInput = (EditText) view.findFocus();
+
+      // 'correction' callback
+      runnable = new Runnable() {
+        @Override
+        public void run() {
+          // set valid minutes && move caret to the end of input
+          view.setCurrentMinute(snapMinutesToInterval(realMinutes));
+          view.setCurrentHour(hourOfDay);
+          textInput.setSelection(textInput.getText().length());
+        }
+      };
+
+      handler.postDelayed(runnable, 500);
+      return;
+    }
+
+    view.setCurrentMinute(snapMinutesToInterval(realMinutes));
+    view.setCurrentHour(hourOfDay);
+  }
+
+  @Override
+  public void onTimeChanged(final TimePicker view, final int hourOfDay, final int minute) {
+    final int realMinutes = getRealMinutes(minute);
+
+    // remove pending 'validation' callbacks if any
+    handler.removeCallbacks(runnable);
+
+    if (minutesAreInvalid(realMinutes)) {
+      correctEnteredMinutes(view, hourOfDay, realMinutes);
+      return;
+    }
+
+    super.onTimeChanged(view, hourOfDay, realMinutes);
+  }
+
+  @Override
+  public void onClick(DialogInterface dialog, int which) {
+    switch (which) {
+      case BUTTON_POSITIVE:
+        final int hours = mTimePicker.getCurrentHour();
+        final int realMinutes = getRealMinutes(mTimePicker.getCurrentMinute());
+        int minutesToCallListenerWith = realMinutes;
+
+        if (pickerIsInTextInputMode() && minutesAreInvalid(realMinutes)) {
+          minutesToCallListenerWith = snapMinutesToInterval(realMinutes);
+        }
+
+        if (mTimeSetListener != null) {
+          mTimeSetListener.onTimeSet(mTimePicker, hours, minutesToCallListenerWith);
+        }
+        break;
+      case BUTTON_NEGATIVE:
+        cancel();
+        break;
+    }
+  }
+
+  @Override
+  public void updateTime(int hourOfDay, int minuteOfHour) {
+    mTimePicker.setCurrentHour(hourOfDay);
+    mTimePicker.setCurrentMinute(snapMinutesToInterval(minuteOfHour));
+  }
+
+  /**
+   * Apply visual style in 'spinner' mode
+   * Adjust minutes to correspond selected interval
+   */
+  @Override
+  public void onAttachedToWindow() {
+    super.onAttachedToWindow();
+    int timePickerId = mContext.getResources().getIdentifier("timePicker", "id", "android");
```

Comment:
again, I'd find it preferable to handle run all the custom code only when we're interested in it (when user sets custom minuteinterval) and otherwise just call super.onAttachedToWindow(); and then return immediately

Reply:
Make sense, I added the check if minute interval is set to run this code